detector/calorimeter/GenericCalEndcap_o2_v01_geo.cpp

```diff
@@ -0,0 +1,244 @@
+//====================================================================
+//  Modified Generic Endcap Driver for the CLIC detector
+//--------------------------------------------------------------------
+//
+//  Author     : N.Nikiforou
+//  Adapted from PolyhedraBarrel Calorimeter by M. Frank
+//  Note: This driver supports conical inner shape in the endcap controlled by rmin and rmin2
+//
+//====================================================================
+#include "DD4hep/DetFactoryHelper.h"
+#include "DDRec/DetectorData.h"
+#include "DDSegmentation/Segmentation.h"
+#include "XML/Layering.h"
+#include "XML/Utilities.h"
+
+using namespace std;
+
+using dd4hep::_toString;
+using dd4hep::BUILD_ENVELOPE;
+using dd4hep::Cone;
+using dd4hep::Detector;
+using dd4hep::DetElement;
+using dd4hep::Layering;
+using dd4hep::Material;
+using dd4hep::PlacedVolume;
+using dd4hep::PolyhedraRegular;
+using dd4hep::Position;
+using dd4hep::Readout;
+using dd4hep::Ref_t;
+using dd4hep::RotationZYX;
+using dd4hep::Segmentation;
+using dd4hep::SensitiveDetector;
+using dd4hep::SubtractionSolid;
+using dd4hep::Transform3D;
+using dd4hep::Volume;
+
+using dd4hep::rec::LayeredCalorimeterData;
+
+// workaround for DD4hep v00-14 (and older)
+#ifndef DD4HEP_VERSION_GE
+#define DD4HEP_VERSION_GE(a, b) 0
+#endif
+
+static Ref_t create_detector(Detector& theDetector, xml_h e, SensitiveDetector sens) {
+  xml_det_t x_det = e;
+  int det_id = x_det.id();
+  string det_name = x_det.nameStr();
+  DetElement sdet(det_name, det_id);
+
+  // --- create an envelope volume and position it into the world ---------------------
+
+  Volume envelope = dd4hep::xml::createPlacedEnvelope(theDetector, e, sdet);
```

Comment:
The envelope doesn't get the conical cutout, does it?

Reply:
No, but it also doesn't seem to get it in the original version.